.github/workflows/coverity_scan.yml

```diff
@@ -0,0 +1,83 @@
+name: Coverity Scan
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '30 0 * * 1'
+jobs:
+  has_changed:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    outputs:
+      changed: ${{ steps.check.outputs.changed }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Print latest commit
+        run: echo ${{ github.sha }}
```

Comment:
Thanks for review, I guest it depends on my overall opinion on readability so I mixed it.This was actually already pushed to master, it was just preset in this PR since I did not rebase it correctly. Only flake8 fixes are now in this PR.